src/language/branches.md

```diff
@@ -0,0 +1,295 @@
+---
+title: Branches 
+description: Learn how to use branches to control the flow of your Dart code.
+---
+
+You can control the flow of your Dart code using branches:
+
+-   `if`, `case`, and `else`
+-   `switch` and `case` statements
+-   `switch` and `case` expressions
+
+Other methods of manipulating control flow in Dart are:
+
+-   [`for` loops][]
+-   [`while` and `do while` loops][]
+-   [`break` and `continue`][break]
+-   [`assert`][]
+
+You can also affect control flow using [Exceptions][], like `try-catch` and `throw`.
+
+## If else
+
+Dart supports `if` statements with optional `else` statements:
+
+<?code-excerpt "misc/lib/language_tour/control_flow.dart (if-else)"?>
+```dart
+if (isRaining()) {
+  you.bringRainCoat();
+} else if (isSnowing()) {
+  you.wearJacket();
+} else {
+  car.putTopDown();
+}
+```
+
+The statement conditions must be expressions
+that evaluate to [boolean values][boolean], nothing else.
+To learn how to use `if` in an expression context, see [Conditional expressions][]
+
+## If case
+
+Dart `if` statements support `case` clauses followed by a [case pattern][pattern]: 
+
+```dart
+if (pair case [int x, int y]) return Point(x, y);
+```
+
+If the case pattern (the list pattern `[int x, int y]` in the example above)
+matches the case expression (`pair`), then the branch (`return Point(x, y)`)
+executes with any variables the pattern defines in scope (`x` and `y`).
+
+Otherwise, the else branch is executed, if there is one:
+
+```dart 
+if (pair case [int x, int y]) {
+  print('Was coordinate array $x,$y');
+} else {
+  throw FormatException('Invalid coordinates.');
+}
+```
+
+If-case is a simple way to match and destructure against a
+_single_ pattern. To test a value against _multiple_ patterns, use [switch](#switch-case).
+
+## Switch case
+
+Switch cases (both [statements](#switch-statements) and [expressions](#switch-expressions))
+in Dart compare an object against one or more [case patterns][pattern]. 
+
+Case patterns allow control flow to
+either match and destructure the object being switched, or continue execution
```

Comment:
Perhaps?:
```suggestion
either match and destructure the object being switched on, or continue execution
```
'Switching on' an object is a rather common phrase. Some people would say that object is the 'scrutinee', but perhaps that's too specialized (it's _the_ thing to say in the functional programming language community).

Reply:
Agreed it's a little too specialized here, but I'll definitely be using "scrutinee" in the functional-style-dart blog post I'm writing!